### PR TITLE
Passive Perks for warrior priests

### PR DESCRIPTION
**Pull request by Z3rca**
Added all passive perks for the warrior priest 
The healing hand prayer have been added as starting prayer (Comet of Sigmar needs laters to be moved, it should be maybe a higher tier perk)
A Generalized Passive mechanic came handy to implement easy generalizable perk logic for explainable numbers all over the code.

### CODE REVIEW

File: CSharpSourceCode/CampaignMechanics/CharacterCreation/TORCharacterCreationContent.cs [hunk 1]
```diff
@@ -213,7 +213,8 @@ private void OnOptionFinalize(TaleWorlds.CampaignSystem.CharacterCreationContent
             if (selectedOption.OptionText == "Priest Acolyte")
             {
                 Hero.MainHero.AddAttribute("Priest");
-                Hero.MainHero.AddAbility("HealingAOE");
+                Hero.MainHero.AddAbility("HealingHand");
+                Hero.MainHero.AddAbility("CometOfSigmar");
```
Comment (hunharibo) [inline CSharpSourceCode/CampaignMechanics/CharacterCreation/TORCharacterCreationContent.cs]: remove comet

File: CSharpSourceCode/CharacterDevelopment/CareerSystem/WarriorPriestCareerChoices.cs [hunk 2]
```diff
@@ -252,35 +258,37 @@ private void InitializeKeyStones()
...
-            _bookOfSigmarPassive3.Initialize(TORCareers.WarriorPriest, "After battle, all critically wounded companions are restored to minimum 20% of their health.", "BookOfSigmar", false, ChoiceType.Passive, null);
-            _bookOfSigmarPassive4.Initialize(TORCareers.WarriorPriest, "Wounded troops in your party heal faster.", "BookOfSigmar", false, ChoiceType.Passive, null);
+            _bookOfSigmarPassive1.Initialize(TORCareers.WarriorPriest, "Increases hitpoints by 10.", "BookOfSigmar", false, ChoiceType.Passive, null,new CareerChoiceObject.PassiveEffect(10, PassiveEffectType.Health));
+            _bookOfSigmarPassive2.Initialize(TORCareers.WarriorPriest, "Higher  troop morale for all troops", "BookOfSigmar", false, ChoiceType.Passive, null,new CareerChoiceObject.PassiveEffect(10, PassiveEffectType.TroopMorale));
+            _bookOfSigmarPassive3.Initialize(TORCareers.WarriorPriest, "After battle, all critically wounded companions restore 20 hitpoints", "BookOfSigmar", false, ChoiceType.Passive, null); // PostBattleCampaignBehavior 30 //TODO REDO , take for other career
```
Comment (hunharibo) [inline CSharpSourceCode/CharacterDevelopment/CareerSystem/WarriorPriestCareerChoices.cs]: left as TODO. Fix?

File: CSharpSourceCode/Extensions/AgentExtensions.cs [hunk 3]
```diff
@@ -74,7 +75,7 @@ public static bool IsUnbreakable(this Agent agent)
             return agent.GetAttributes().Contains("Unbreakable");
         }
 
-        public static bool IsHuman(this Agent agent)
+        public static bool IsHuman(this Agent agent)            //I think this is not really helpful
         {
             return agent.GetAttributes().Contains("Human");
         }
```
Comment (hunharibo) [inline CSharpSourceCode/Extensions/AgentExtensions.cs]: remove completely. Not needed/used.

File: CSharpSourceCode/HarmonyPatches/DamagePatch.cs [hunk 4]
```diff
@@ -129,26 +138,30 @@ public static bool PreHandleBlow(ref Blow b, Agent __instance)
...
+                    damageCategories[i] *= 1 + damageAmplifications[i];
                     resultDamage += (int)damageCategories[i];
                 }
             }
-            resultDamage = (int)(resultDamage * wardSaveFactor * (1 + damagePercentages[(int)DamageType.All]));
+            resultDamage = (int)(resultDamage * wardSaveFactor * (1 + damageAmplifications[(int)DamageType.All])); 
             var originalDamage = b.InflictedDamage;
             b.InflictedDamage = resultDamage;
             b.BaseMagnitude = resultDamage;
-            if (victim.GetAttributes().Contains("Unstoppable")) b.BlowFlag |= BlowFlags.ShrugOff;
```
Comment (hunharibo) [inline CSharpSourceCode/HarmonyPatches/DamagePatch.cs:141]: revert this change pls. We are depending on the Unstoppable attribute.

File: CSharpSourceCode/Models/TORAgentStatCalculateModel.cs [hunk 5]
```diff
@@ -316,7 +318,55 @@ public AgentPropertyContainer AddPerkEffectsToAgentPropertyContainer(Agent agent
...
+            }
+            else if (agentLeader !=null&&agentLeader==CharacterObject.PlayerCharacter)
+            {
+                if (!Agent.Main.GetHero().HasAnyCareer()) return result;
+                
+                var choices = Agent.Main.GetHero().GetAllCareerChoices();
+
+                if (choices.Contains("RelentlessFanaticPassive3")&& mask == PropertyMask.Defense&&attackMask==AttackTypeMask.Ranged)
+                {
+                    if (agent.Character.StringId == "tor_empire_flagellant")
```
Comment (hunharibo) [inline CSharpSourceCode/Models/TORAgentStatCalculateModel.cs]: rework passive so it applies for all religious units, not just the flagellants. It is not good design practice to hardcode troop IDs.

**Review by hunharibo — CHANGES_REQUESTED** [review]

File: CSharpSourceCode/CharacterDevelopment/CareerSystem/WarriorPriestCareerChoices.cs (hunk 2 above)
Comment (Z3rca) [reply CSharpSourceCode/CharacterDevelopment/CareerSystem/WarriorPriestCareerChoices.cs]: will be changed later with more careers

**Review by hunharibo — APPROVED** [review]
